### PR TITLE
add ability to set time interval to 'cut' measures

### PR DESCRIPTION
**Pull request by Almouro**

### CODE REVIEW

File: packages/web-reporter/writeReport.ts [hunk 1]
```diff
@@ -1,11 +1,44 @@
...
+export const getMeasuresForTimeInterval = ({
+  duration,
+  skip,
+  results,
+}: {
+  duration: number | null;
+  skip: number;
+  results: TestCaseResult[];
+}): TestCaseResult[] => {
+  const firstMeasureIndex = skip / MEASURE_INTERVAL;
```
Comment (MaeIg) [inline packages/web-reporter/writeReport.ts:16]: If skip === 300, the first index will be 0.6.

We should use Math.ceil on the result

File: packages/web-reporter/writeReport.ts [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
+
+  return duration
+    ? results.map((result) => ({
+        ...result,
+        iterations: result.iterations.map((iteration) => ({
+          ...iteration,
+          measures: iteration.measures.slice(
+            firstMeasureIndex,
+            firstMeasureIndex + duration / MEASURE_INTERVAL + 1
```
Comment (MaeIg) [inline packages/web-reporter/writeReport.ts:25]: If duration === 300, the first index will be firstIndex + 1.6.

We should use Math.ceil on the result

File: packages/web-reporter/writeReport.ts [hunk 3]
```diff
@@ -1,11 +1,44 @@
...
+  skip,
+  results,
+}: {
+  duration: number | null;
+  skip: number;
+  results: TestCaseResult[];
+}): TestCaseResult[] => {
+  const firstMeasureIndex = skip / MEASURE_INTERVAL;
+
+  return duration
```
Comment (MaeIg) [inline packages/web-reporter/writeReport.ts:18]: We should be able to skip some frames even if duration is null.

For example, I want to skip the first 500ms but keep all the data after

File: packages/web-reporter/writeReport.test.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  // We're just mocking measure to make tests more readable here
+  return name as unknown as Measure;
+};
+
+const mockResultIteration = (name: string[]) => ({
+  measures: name.map(mockMeasure),
+  time: 0,
+});
+
+test("getMeasuresForTimeInterval", () => {
```
Comment (MaeIg) [inline packages/web-reporter/writeReport.test.ts:14]: Maybe add tests:
  - if duration is null 
  - if skip or duration is not a multiple of 500

File: packages/web-reporter/writeReport.ts [hunk 5]
```diff
@@ -1,11 +1,55 @@
 import fs from "fs";
+import { TestCaseResult } from "@perf-profiler/types";
+
+// We should have better management of the time interval over the repo
+const MEASURE_INTERVAL = 500;
+
+const assertTimeIntervalMultiple = (n: number) => {
+  if (n % MEASURE_INTERVAL !== 0) {
+    throw new Error(
```
Comment (Almouro) [inline packages/web-reporter/writeReport.ts:9]: I think at this point it makes more sense to do this, as skipping 600ms for instance would not make the report start at 600ms but at 1000ms

File: packages/web-reporter/writeReport.test.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      getMeasuresForTimeInterval({
+        duration: null,
+        skip: 1000,
+        results: [RESULT],
+      })
+    ).toEqual([
+      {
+        iterations: [
+          mockResultIteration(["ITERATION1_1000_ms"]),
+          mockResultIteration([]),
```
Comment (Almouro) [inline packages/web-reporter/writeReport.test.ts:74]: Granted, we could handle this case a bit better

This whole feature is mostly useful if iterations have the same duration tbf

File: packages/web-reporter/writeReport.ts (hunk 5 above)
Comment (MaeIg) [reply packages/web-reporter/writeReport.ts:9]: It's ok for me :)

**Review by MaeIg — APPROVED** [review]

**Pull request merged**